velox/functions/prestosql/tests/ArrayPositionTest.cpp

```diff
@@ -0,0 +1,441 @@
+        "array_position(c0, c1)",
+        makeRowVector({
+            arrayVector,
+            makeConstant(search, arrayVector->size()),
+        }));
+
+    assertEqualVectors(makeNullableFlatVector<int64_t>(expected), result);
+  };
+
+  testPosition("red", {1, 0, 0, 0, 1});
+  testPosition("blue", {2, 0, 0, 0, 0});
+  testPosition("yellow", {0, 2, 0, 0, 0});
+  testPosition("green", {0, 0, 0, 0, 3});
+  testPosition("crimson red", {0, 0, 0, 0, 0});
+  testPosition(
+      std::nullopt,
+      {std::nullopt, std::nullopt, std::nullopt, std::nullopt, std::nullopt});
+}
+
+TEST_F(ArrayPositionTest, boolean) {
+  auto arrayVector = makeNullableArrayVector<bool>({
+      {true, false},
+      {true},
+      {false},
+      {},
+      {std::nullopt},
+      {true, std::nullopt, true},
+      {false, false, false},
+  });
+
+  auto testPosition = [&](std::optional<bool> search,
+                          const std::vector<std::optional<int64_t>>& expected) {
+    auto result = evaluate<SimpleVector<int64_t>>(
+        "array_position(c0, c1)",
+        makeRowVector({
+            arrayVector,
+            makeConstant(search, arrayVector->size()),
+        }));
+
+    assertEqualVectors(makeNullableFlatVector<int64_t>(expected), result);
+  };
+
+  testPosition(true, {1, 1, 0, 0, 0, 1, 0});
+  testPosition(false, {2, 0, 1, 0, 0, 0, 1});
+  testPosition(
+      std::nullopt,
+      {std::nullopt,
+       std::nullopt,
+       std::nullopt,
+       std::nullopt,
+       std::nullopt,
+       std::nullopt,
+       std::nullopt});
+}
+
+TEST_F(ArrayPositionTest, row) {
+  std::vector<std::vector<variant>> data = {
+      {
+          variant::row({1, "red"}),
+          variant::row({2, "blue"}),
+          variant::row({3, "green"}),
+      },
+      {
+          variant::row({2, "blue"}),
+          variant(TypeKind::ROW), // null
+          variant::row({5, "green"}),
+      },
+      {},
+      {
+          variant(TypeKind::ROW), // null
+      },
+      {
+          variant::row({1, "yellow"}),
+          variant::row({2, "blue"}),
+          variant::row({4, "green"}),
+          variant::row({5, "purple"}),
+      },
+  };
+
+  auto rowType = ROW({INTEGER(), VARCHAR()});
+  auto arrayVector = makeArrayOfRowVector(rowType, data);
+
+  auto testPosition = [&](int32_t n,
+                          const char* color,
+                          const std::vector<std::optional<int64_t>>& expected) {
+    auto search =
+        makeConstantRow(rowType, variant::row({n, color}), arrayVector->size());
+
+    auto result = evaluate<SimpleVector<int64_t>>(
+        "array_position(c0, c1)", makeRowVector({arrayVector, search}));
+
+    assertEqualVectors(makeNullableFlatVector<int64_t>(expected), result);
+  };
+
+  testPosition(1, "red", {1, 0, 0, 0, 0});
+  testPosition(2, "blue", {2, 1, 0, 0, 2});
+  testPosition(4, "green", {0, 0, 0, 0, 3});
+  testPosition(5, "green", {0, 3, 0, 0, 0});
+  testPosition(1, "purple", {0, 0, 0, 0, 0});
+}
+
```

Comment:
Can we also have fn's to search for arrays or maps ?

Reply:
Added tests for array of arrays and array of maps.